internal/services/managedapplications/managed_application_definition_resource.go

```diff
@@ -193,105 +190,99 @@ func resourceManagedApplicationDefinitionRead(d *pluginsdk.ResourceData, meta in
 			d.SetId("")
 			return nil
 		}
-		return fmt.Errorf("failed to read Managed Application Definition %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+		return fmt.Errorf("failed to read Managed Application Definition %s: %+v", id, err)
 	}
 
-	d.Set("name", id.Name)
-	d.Set("resource_group_name", id.ResourceGroup)
-	d.Set("location", location.NormalizeNilable(resp.Location))
+	if m := resp.Model; m != nil {
+		p := m.Properties
+
+		d.Set("name", m.Name)
+		d.Set("resource_group_name", id.ResourceGroupName) // missing from response?
+		d.Set("location", location.NormalizeNilable(m.Location))
 
-	if props := resp.ApplicationDefinitionProperties; props != nil {
-		if err := d.Set("authorization", flattenManagedApplicationDefinitionAuthorization(props.Authorizations)); err != nil {
+		if err := d.Set("authorization", flattenManagedApplicationDefinitionAuthorization(p.Authorizations)); err != nil {
 			return fmt.Errorf("setting `authorization`: %+v", err)
 		}
-		d.Set("description", props.Description)
-		d.Set("display_name", props.DisplayName)
-		d.Set("package_enabled", props.IsEnabled)
-		d.Set("lock_level", string(props.LockLevel))
-	}
+		d.Set("description", p.Description)
+		d.Set("display_name", p.DisplayName)
+		d.Set("package_enabled", p.IsEnabled)
+		d.Set("lock_level", string(p.LockLevel))
+
+		// the following are not returned from the API so lets pull it from state
+		if v, ok := d.GetOk("create_ui_definition"); ok {
+			d.Set("create_ui_definition", v.(string))
+		}
+		if v, ok := d.GetOk("main_template"); ok {
+			d.Set("main_template", v.(string))
+		}
+		if v, ok := d.GetOk("package_file_uri"); ok {
+			d.Set("package_file_uri", v.(string))
+		}
 
-	// the following are not returned from the API so lets pull it from state
-	if v, ok := d.GetOk("create_ui_definition"); ok {
-		d.Set("create_ui_definition", v.(string))
-	}
-	if v, ok := d.GetOk("main_template"); ok {
-		d.Set("main_template", v.(string))
-	}
-	if v, ok := d.GetOk("package_file_uri"); ok {
-		d.Set("package_file_uri", v.(string))
+		return tags.FlattenAndSet(d, m.Tags)
 	}
 
-	return tags.FlattenAndSet(d, resp.Tags)
+	return nil
 }
 
 func resourceManagedApplicationDefinitionDelete(d *pluginsdk.ResourceData, meta interface{}) error {
 	client := meta.(*clients.Client).ManagedApplication.ApplicationDefinitionClient
 	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.ApplicationDefinitionID(d.Id())
+	id, err := applicationdefinitions.ParseApplicationDefinitionID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	future, err := client.Delete(ctx, id.ResourceGroup, id.Name)
+	resp, err := client.Delete(ctx, *id)
 	if err != nil {
-		return fmt.Errorf("failed to delete Managed Application Definition %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
-	}
+		if response.WasNotFound(resp.HttpResponse) {
+			return nil
+		}
```

Comment:
this masks bugs and should be removed across (almost) all instances in the provider fwiw:

```suggestion
```

From Terraform's side if it's gone it'll be detected during the Read function, which gets called before Delete, meaning if it's been deleted in the remote API then Terraform won't call the Delete function at all - so this is safe to remove.

There's a handful of API's where this isn't the case, but I'm fairly confident this isn't one of them (and IIRC the new base layer accounts for this during deletions automatically)